From 2dab4c4e81648f47d7e58ff3f77f3caf7860059c Mon Sep 17 00:00:00 2001
From: DRC <davide@licheni.net>
Date: Thu, 6 Apr 2017 00:25:08 +0200
Subject: [PATCH 4/5] fix custom view mode on ordering list

respect the option of custom view mode other than default and VIEW_MODE_TITLE
---
 src/Plugin/Field/FieldWidget/EntityReferenceDragDropWidget.php | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/Plugin/Field/FieldWidget/EntityReferenceDragDropWidget.php b/src/Plugin/Field/FieldWidget/EntityReferenceDragDropWidget.php
index 5ad6323..d4512a9 100644
--- a/src/Plugin/Field/FieldWidget/EntityReferenceDragDropWidget.php
+++ b/src/Plugin/Field/FieldWidget/EntityReferenceDragDropWidget.php
@@ -244,7 +244,11 @@ class EntityReferenceDragDropWidget extends OptionsWidgetBase implements Contain
    * @return array
    */
   protected function optionsToRenderableArray(array $options, $key, $list_title, array $classes = array(), array $wrapper_classes = array()) {
-    $view_mode = $this->getSetting('view_mode');
+    // view mode is of the form 'node.full_content'
+    $view_mode_name = $this->getSetting('view_mode');
+    $view_mode_name = explode('.', $view_mode_name);
+    $view_mode = end($view_mode_name);
+
     $target_type_id = $this->fieldDefinition->getFieldStorageDefinition()->getSetting('target_type');
     $items = array();
     $entities = array();
@@ -262,7 +266,7 @@ class EntityReferenceDragDropWidget extends OptionsWidgetBase implements Contain
         ),
       );
       if ($view_mode !== static::VIEW_MODE_TITLE) {
-        $item += $this->entityTypeManager->getViewBuilder($target_type_id)->view($entities[$id]);
+        $item += $this->entityTypeManager->getViewBuilder($target_type_id)->view($entities[$id], $view_mode);
       }
       else {
         $item += array(
-- 
2.11.0

